### DISCUSSION TITLE
Best way to collaborate and contribute to the code

### DISCUSSION THREAD

**Discussion by aranhax** [iiasa/CWatM#159] [Q&A]
Hello all,

I've been using CWatM to model streamflow in Canada. Canada streamflow can be heavily driven by snow. When using the `snowmelt_radiation`  option in CWatM, the implementation uses a radiation part from evaporationPot where snowmelt has a temperature part and a radiation part. I am trying to implement on CWatM the energy balance in the `snow_frost.py`  which will also include turbulent exchange and heat from rain. What would be the best way to contribute to the code? Are there some guidelines described somewhere to follow? Code standards or tests? 

Thank you in advance.

**Comment by abbylute** [comment]
I am also curious about this. Would it make more sense to integrate new processes directly in CWatM? Or perhaps, for groups of processes would it make sense to couple CWatM with a separate model that includes the new components (similar to the approach used for OGGM)? I'd appreciate any insights or suggestions on this.

**Reply by dof1985** [reply]
Dear @abbylute ,

Please see the recent paper by Sarah Hanus about the coupling of CWatM and OGGM.

[https://gmd.copernicus.org/articles/17/5123/2024/](https://gmd.copernicus.org/articles/17/5123/2024/)

**Reply by abbylute** [reply]
Thank you Dor. I am familiar with that paper. So it sounds like the preference for larger bodies of code would be to couple them to CWatM, not to integrate into CWatM directly. Thanks!

**Comment by dof1985** [comment]
It largely depends on how would you like the models to interact. 

If you just need a soft coupling transffering inputs-outputs  (usually unidirectional) the example with OGGM would be sufficient. 

A hard coupling approach is usually required if you are looking for an on-going interaction, e.g., in the case of the Modflow coupling:
[https://gmd.copernicus.org/articles/15/7099/2022/gmd-15-7099-2022.html](https://gmd.copernicus.org/articles/15/7099/2022/gmd-15-7099-2022.html)

Alternatively, you can borrow specific modules from a bigger model and integrate them directly into CWatM's code.

Best
Dor
